src/resolver.js

```diff
@@ -70,19 +70,33 @@ function resolve(link, ctx, visited = []) {
         ctx.result.push(createYAMLParseError(e, ctx, resolvedFilePath));
         return { node: undefined };
       }
-      // FIXME: lost yaml parsing and file read errors here
     } else if (isFullyQualifiedUrl(resolvedFilePath)) {
       try {
-        const xhr = new XMLHttpRequest();
-        xhr.open('GET', resolvedFilePath, false);
-        xhr.send();
+        ctx.redoclyClient.processRegistryDependency(resolvedFilePath, ctx);
+
+        const headers = {};
+        for (let i = 0; i < ctx.headers.length; i++) {
+          if (ctx.headers[i].regexp.test(resolvedFilePath)) {
+            headers[ctx.headers[i].name] = ctx.headers[i].value;
+          }
+        }
+
+        const req = await fetch(resolvedFilePath, { headers });
 
-        if (xhr.status !== 200) {
+        if (!req.ok) {
+          // FIXME: lost file read errors here
+          // ctx.result.push(...);
```

Comment:
Could you add an error to result here?
Right now it will just silently fail e.g. for 404 links.